[Fix] Release KHR_texture_basisu decoder promise after texture upload

FlorentinMonteil opened the pull request:
The decode promise from basisu extension is keeping reference to decoded buffer, leading to (maybe) useless CPU heap memory retention. 
It seems that setupTexture is called only once per element and this promise is not used or reused outside the setupFunction, setting this to null effectively free the memory from the cpu.

FlorentinMonteil committed:
[Fix] Release KHR_texture_basisu decoder promise after texture upload

plepers commented:
Generally, gltf.allocateGl() methods don't destroy gpu resources afterward. This way you can quickly call it again on another context, in case the context is lost for example.
Maybe you can add an option on the extension to prevent decoded texture to be kept: 
`Gltf.addExtension(new KHR_texture_basisu( decoder, { keepDecoded:false });`

FlorentinMonteil committed:
update: KHR_texture_basisu options

Add keepDecoded options to KHR_texture_basisu, if false will release the decoded promise after setup

FlorentinMonteil commented:
@plepers updated :)

plepers commented:
Thx!
